src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -1608,6 +1620,13 @@ fn build_script_local_fingerprints(
     bool,
 ) {
     assert!(unit.mode.is_run_custom_build());
+    let envs = build_runner
+        .bcx
+        .target_data
+        .info(unit.kind)
+        .get_target_envs()
+        .unwrap()
+        .clone();
```

Comment:
I've not looked at the full control flow.  Do we have to `clone`?

Reply:
Ah, `envs` gets captured by a closure that this function returns so everything has to be `'static`.